File: CAT 1/employee.cpp
#include <iostream>
#include <string>
using namespace std;

class Employee {
private:
    string employeeName;
    int employeeId;
    double employeeSalary;

public:
    
    Employee() : employeeName("unknown"), employeeId(0), employeeSalary(0.00) {}

    Employee(const string& n, int id, double s) : employeeName(n), employeeId(id), employeeSalary(s) {}


    string getEmployeeName() const {
        return employeeName;
    }

    int getEmployeeId() const {
        return employeeId;
    }

    double getEmployeeSalary() const {
        return employeeSalary;
    }

    
    void setEmployeeName(const string& n) {
        employeeName = n;
    }

    void setEmployeeId(int id) {
        employeeId = id;
    }

    void setEmployeeSalary(double s) {
        employeeSalary = s;
    }

    // Method to raise salary
    double raiseSalary(double increment) {
        employeeSalary += increment;
        return employeeSalary;
    }
};

int main() {
    // Create an instance using no-args constructor
    Employee e1;
    e1.raiseSalary(5000.00);

    // Print out e1's details
    cout << "Employee ID: " << e1.getEmployeeId() << std::endl;
    cout << "Employee Name: " << e1.getEmployeeName() << std::endl;
    cout << "Employee Salary: " << e1.getEmployeeSalary() << std::endl;

    
    Employee e2("Nelson", 100, 20000.00);

    cout << "Employee Name: " << e2.getEmployeeName() << std::endl;
    cout << "Employee Salary: " << e2.getEmployeeSalary() << std::endl;

    // Using setters/mutators
    Employee e3;
    e3.setEmployeeName("John");
    e3.setEmployeeSalary(40000.00);
    e3.setEmployeeId(300);

    // Print e3's details
    cout << "Employee ID: " << e3.getEmployeeId() << std::endl;
    cout << "Employee Name: " << e3.getEmployeeName() << std::endl;
    cout << "Employee Salary: " << e3.getEmployeeSalary() << std::endl;

    return 0;
}

File: CAT 1/Employee.py
class Employee:
    def __init__(self, name="unknown", emp_id=0, salary=0.00):
        self.employee_name = name
        self.employee_id = emp_id
        self.employee_salary = salary

    # Getter methods
    def get_employee_name(self):
        return self.employee_name

    def get_employee_id(self):
        return self.employee_id

    def get_employee_salary(self):
        return self.employee_salary

    # Setter methods
    def set_employee_name(self, name):
        self.employee_name = name

    def set_employee_id(self, emp_id):
        self.employee_id = emp_id

    def set_employee_salary(self, salary):
        self.employee_salary = salary

    # Method to raise salary
    def raise_salary(self, increment):
        self.employee_salary += increment
        return self.employee_salary


# Test cases
if __name__ == "__main__":
    # Create an instance using no-args constructor
    e1 = Employee()
    e1.raise_salary(5000.00)

    # Print out e1's details
    print("Employee ID:", e1.get_employee_id())
    print("Employee Name:", e1.get_employee_name())
    print("Employee Salary:", e1.get_employee_salary())

    # Create an instance using parameterized constructor
    e2 = Employee("Nelson", 100, 20000.00)

    # Print out e2's details
    print("Employee ID:", e2.get_employee_id())
    print("Employee Name:", e2.get_employee_name())
    print("Employee Salary:", e2.get_employee_salary())

    # Use setters/mutators
    e3 = Employee()
    e3.set_employee_name("John")
    e3.set_employee_salary(40000.00)
    e3.set_employee_id(300)

    # Print e3's details
    print("Employee ID:", e3.get_employee_id())
    print("Employee Name:", e3.get_employee_name())
    print("Employee Salary:", e3.get_employee_salary())